apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryInsertRow.cs

```diff
@@ -197,54 +197,42 @@ internal TableDataInsertAllRequest.RowsData ToRowsData(bool allowEmptyInsertIds)
 
         internal Dictionary<string, object> GetJsonValues() => _fields.ToDictionary(pair => pair.Key, pair => ConvertRowValue(pair.Value));
 
-        private static object ConvertRowValue(object value)
+        private static object ConvertRowValue(object value) => value switch
         {
-            // Types that require special handling
-            switch (value)
+            null => null,
+            DateTime dt => dt.Kind switch
```

Comment:
I think I'd like better 
```suggestion
            DateTime dt when dt.Kind == DateTimeKind.Unspecified => dt.ToString...,
            DateTime dt when dt.Kind == DateTimeKind.Utc = > ...,
            DateTime dt => throw ...
```
and similar for the TimeStamp down belown, to avoid the nesting.

I'll let @jskeet give his opinion though.

Reply:
Sure Amanda. Done as per suggestion from Jon.